File: cpp/prob_867.cpp
class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& A) {
        int N_rows = A.size();
        int N_cols = A[0].size();
        # If A is a square matrix, transpose in place
        if (N_rows == N_cols) {
            for (int row_i=0; row_i<N_rows; row_i++) {
                for (int col_i=row_i+1; col_i<N_rows; col_i++) {
                    int temp = A[row_i][col_i];
                    A[row_i][col_i] = A[col_i][row_i];
                    A[col_i][row_i] = temp;
                };
            };
            return A;
        }
        # otherwise create new matrix
        else {
            vector<vector<int>> A_T;
            for (int col_i=0; col_i<N_cols; col_i++) {
                vector<int> row_T;
                for (int row_i=0; row_i<N_rows; row_i++) {
                    row_T.push_back(A[row_i][col_i]);
                };
                A_T.push_back(row_T);
            };
            return A_T;
        };
    };
};

File: py/prob_867.py
class Solution:
    def transpose(self, A):
        """
        :type A: List[List[int]]
        :rtype: List[List[int]]
        """
        N_rows = len(A)
        N_cols = len(A[0])
        # If A is a square matrix, transpose in place
        if N_rows == N_cols:
            for row_i in range(N_rows):
                for col_i in range(row_i+1, N_rows):
                    temp = A[row_i][col_i]
                    A[row_i][col_i] = A[col_i][row_i]
                    A[col_i][row_i] = temp
            return A
        # otherwise create new matrix
        else:
            A_T = []
            for i in range(N_cols):
                row_T = []
                for row in A:
                    row_T.append(row[i])
                A_T.append(row_T)
            return A_T